### DISCUSSION TITLE
How to add HTTPS directly in container ?

### DISCUSSION THREAD

**Discussion by gamersalpha** [tteck/Proxmox#2244] [Q&A] [answered]
Hello there, 

can u have explaination how to add https acces on the container please ?

heimdell ?
LinkWarden ?
thanks all

**Comment by AlexGeek77** [comment, marked as answer]
NGIX proxy manager if you're looking for https remote access to your network. If you want each to have their own https service then you'll need to work getting certificates going on each service.

**Comment by gamersalpha** [comment]
Hello thanks, 

but i need a full https from the serveur : 

not a https <--> HTTP communication

https <--> Nginx Proxy manager <--> http service (Heimdell, linkwarden)

i want full node access https, no http on lan ?

thanks all

**Comment by Vakrehus** [comment]
NPM is the way to go:

`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/nginxproxymanager.sh)"`

**Reply by gamersalpha** [reply]
but even i install we keep an acces in http betwin nginx proxy manager and linkwarden

this is an issue ?

**Comment by gamersalpha** [comment]
Ok so i used HA PROXY and works, but i m no sure of the https <---HA PROXY --> Heimdell

hope some one can explain me how fix his security issues
